### DISCUSSION TITLE
Duplicate deck when upgrading from v4.0 to v4.1

### DISCUSSION THREAD

**Discussion by axelboc** [axelboc/anki-ultimate-geography#390] [Q&A] [answered]
Somebody on Reddit [reported an issue with upgrading from v4.0 to v4.1](https://www.reddit.com/r/Anki/comments/ldw0op/ultimate_geography_v41_released/gmazhyw/). Any idea what could have gone wrong?

> I followed the instructions in the section labelled "Other languages and versions" (which is where the instructions send you under "Upgrading / Minor version"). I installed the CrowdAnki add-on. I downloaded the 4.1 ZIP file. I extracted it. I selected CrowdAnki: Import from disk and did the import (without changing any of the import options).
>
> Now I have *two* decks:
> - Ultimate Geography
> - Ultimate Geography_2
>
> The second one seems to be 4.1, and it doesn't seem to know about any of progress, or any of the options changes I made for 4.0.

**Comment by aplaice** [comment]
That's pretty weird.

> Now I have two decks:
>
> * Ultimate Geography
> * Ultimate Geography_2
>
> The second one seems to be 4.1, and it doesn't seem to know about any of progress, or any of the options changes I made for 4.0.

If they imported the original (4.0) deck via the `.apkg`, then the fact that they get two decks is expected — the `.apkg` export does not contain the `crowdanki_uuid` of the deck, so `CrowdAnki` doesn't and can't know that the newly imported deck is the same as the existing one (named *Ultimate Geography*), so it tries to create a deck named *Ultimated Geography*, fails (since *Ultimate Geography* is taken by the existing deck) and hence creates a deck called *Ultimate Geography_2*.  (Notes do/should "match up" between the `.apkg` and the `CrowdAnki` json, since the note `guid`s *are* present in the `.apkg`.)

The fact that the second deck (*Ultimate Geography_2*) does not contain any of the options changes also makes sense. (`CrowdAnki` forces its `deck_config` onto you, as you @axelboc have already complained :))

(FWIW I think that you can get partially around these two problems by selecting the "Do Not Move Existing Cards" option in the `CrowdAnki` import.)

However, as mentioned above, the notes (and progress) definitely should have all moved to the second deck...

(I've tested several configurations on clean profiles:

1. importing 4.0 `.apkg`, then 4.1 EN zip
2. importing 4.0 `.apkg`, then 4.1 EN extended zip
3. importing 4.0 EN zip, then 4.1 EN zip

and none of them results in the reported behaviour.)

**Reply by axelboc** [reply]
I had no clue about this deck duplication behaviour. I thought the UUID matched between the JSON and APKG files. Perhaps we should find the ideal process (re _Do Not Move Existing Cards_) and mention it in the `README`? Intuitively, I'd say that it depends whether the deck has lost or gained cards...

**Reply by aplaice** [reply]
It seems that I was completely wrong about the UUIDs (Deck, Deck Config and Note type) not being present in the APKG file itself! Inspecting the `collection.anki2` within the APKG, all three are. Nevertheless, only the Note type UUID (and the note GUIDs (but that wasn't in any doubt)) actually gets imported into Anki, when you import the APKG. I've now only tested it in the latest version of Anki (2.1.39), but I remember the behaviour of a duplicate deck being created in far earlier ones as well.

(The 3 UUIDs (for Deck, Deck Config and Note type) are all CrowdAnki "extensions", so the fact that a vanilla APKG export/import won't preserve them is not too unexpected. (Only the note GUID is built-into Anki.))

See also: https://github.com/Stvad/CrowdAnki/issues/6

## Workflow

Once somebody has imported via CrowdAnki once, (and ideally moved any remaining cards they want to keep in the old "APKG" deck to the new duplicate "CrowdAnki" deck, deleted their old "APKG" deck and renamed the new "CrowdAnki" deck), then no duplicate decks will be created on further CrowdAnki exports. (If they don't delete the "original" deck, then at least no _further_ duplicate decks will be created.)  Hence, it's not terribly bad...

However, we definitely should consider whether we should recommend using the _Do Not Move Existing Cards_ option and whether we should add guidance/reassurance about the creation of a "duplicate" deck. (Should we open an issue about it?)

In an ideal world, `CrowdAnki` would detect cases where there is such a situation and ask the user what to do about it.

Edit: In an ideal, ideal world, Anki would fully import all UUIDs...

**Comment by jeprecated** [comment]
I see you have linked the user to this thread, perhaps they can give us some further clues to work with, in the form of an export of their current deck. If the note guids do not match (due to some rogue add-on or action the user has taken in the past) then that would explain this and there is no issue with this deck or CrowdAnki. If they do match, then we can hopefully recreate the issue with their data.

Otherwise I'd do a quick sanity check and ask: *Are you sure the cards were duplicated?* :sweat_smile: Perhaps they opened the new deck, looked at a random card, which happened to be one of the new cards added in 4.1, and assumed the rest of the deck was also empty? :thinking: Long shot, but worth checking before we all spend a lot of time investigating (though nothing can stop @aplaice! :smile:)

**Reply by FiveRedRocks** [reply]
I'm happy to do whatever I can to help figure this out. (And it's certainly possible it's my error -- I'm very new to Anki.)

Please give me specific directions and I'll try to follow them -- thanks.

**Comment by FiveRedRocks** [comment, marked as answer]
OK, now I'm sure this is my mistake. Today I went to study from my "old" Ultimate Geography deck and was astonished to see it showing me French words. Eventually I figured out that I must have accidentally added 6 cards to the original Ultimate Geography, rather than my private, custom French deck. In fact, I now realize that these 6 cards are the *only* cards in Ultimate Geography.

Also, when I click on "stats" in Ultimate Geography_2, it shows me several weeks of my stats, as if I'd been using it all along. So maybe everything's fine, and I can just use (and rename) Ultimate Geography_2. I'm still confused as to why it had me study so few cards yesterday and today, because I hit "again" for 52 cards the last time I studied the original UG before updating. But this is probably just not me understanding the Anki algorithm.

Sorry for the unnecessary bother!

**Reply by jeprecated** [reply]
Haha, no worries! :sweat_smile: It is rather confusing that the deck gets made twice :disappointed: 

> I'm still confused as to why it had me study so few cards yesterday and today, because I hit "again" for 52 cards the last time I studied the original UG before updating. But this is probably just not me understanding the Anki algorithm.

52 wrong cards in a day (out of perhaps 60-100 cards?) sounds like a lot of new cards per day! :scream: A tool like Anki is difficult for human brains to get used to as it is a lagging measure - one that the results of your work only show up later on (sometimes much later!). Put another way, you may struggle for a month with European capitals for a month or two and then find that you just 'get' most of them :+1: however there is a danger to this lagging measure that you should beware: *doing too many reviews* :scream: doing 10 new cards a day does not equate to doing 10 reviews the next day (for a total of 20). In reality you'll eventually get to around 40 cards per day in total, depending on your settings. So really beware how many new cards you decide to take on, and how that will eventually affect your total load. Try the add-on [Anki Simulator](https://ankiweb.net/shared/info/817108664) to get a better overview of this :+1: Don't push yourself to do so much that you will eventually be overwhelmed, it's a long term battle! If you find that you need less cards today or for the next while first lower your new cards, then cap your reviews if need be, for the best results with minimum cards.

I wrote a shitty blogpost to help some friends understand the basics of Anki, perhaps it may help you too: https://ohare.blog/anki-beginner-steps :+1: 

And finally here's a comment about mindset [I said elsewhere](https://www.reddit.com/r/Anki/comments/dlhufp/how_to_deal_with_interference_ultimate_geography/) that may be of help too :+1: 

> More importantly remember why you are doing flashcards! ~~You want to show off to your friends/familt/colleagues and by sounding surpringly knowledgable and worldly, winning a few pub quizes in the process!~~ You are trying to remember some information that you presumably knew almost nothing of beforehand. Without flashcards you would still know that, so anything above there is an improvement.
> 
> That said, our stupid brains are like leaky sieves: **You will get many many flashcards wrong!** Accept that as inevitable and know that every flashcard you get wrong isn't a failure in yourself, but simply Anki highlighting: "Hey, you forgot this! Without me here to remind you it would be gone forever, now you can have it back and remember for a bit longer". Aim for getting 90% of your cards correct (so the manual suggests) as anything above that requires too increasingly more and more reviews with diminishing returns. You can check your percentage in the stats. If you go above or below that, there are ways to change your card frequency (mainly by editting the Interval Modifier).
> 
> Stay positive, and keep learning the world! Feel free to reply/PM with any further questions, thoughts, ~~geography challenges~~, etc.

Best of luck in your fight against your treacherous brain! :brain:

**Reply by FiveRedRocks** [reply]
Thanks for all the great information!

I've set Anki to give me 5 new cards a day, because my life is so busy (and because recognizing the flag of the Faroe Islands doesn't directly impact my livelihood). Recently, my life was so busy that I couldn't use Anki at all for 7 days.

On the 8th day, Anki had me do 48 reviews, all of which I got right. On the 9th day, 89 reviews, of which 19 were "relearning". On the 10th day, when I was super-tired, 101 reviews, of which 51 were relearning.

On the 11th day, I downloaded Ultimate Geography 4.1 and had the confusion documented above. Part of my confusion was the fact that I then only had to do 50 reviews, of which 8 were relearning. Somehow I'd thought that, if I made so many mistakes the day before, I'd get more cards, not fewer.

**Reply by jeprecated** [reply]
Sounds to me like you have an upper limit of reviews set for the deck (this can be set by default too). Open the deck options with the cog on the right of the deck and check if there is a limit of 50 set in the reviews section. If so, up it to whatever you want! 😁 I have all of mine set to 1000 (though I hardly go over 30 haha)

**Reply by FiveRedRocks** [reply]
My upper limit is 9999 reviews in a day, which thankfully I've never reached.
